src/sparsezoo/analysis/Analysis.py

```diff
@@ -0,0 +1,221 @@
+    )
+    num_nodes: int = Field(description="Total number of nodes")
+    num_layers: int = Field(description="Total number of layers")
+    num_operations: int = Field(
+        description="Total number of operations (nodes that are not layers)"
+    )
+    average_sparsity: float = Field(
+        description="Average sparsity across all trainable parameters "
+        "(excluding biases)"
+    )
+    average_four_block_sparsity: float = Field(
+        description="Average sparsity four block sparsity across all "
+        "trainable parameters (excluding biases)"
+    )
+    num_sparse_layers: int = Field(description="Number of layers with any sparsity")
+    num_quantized_layers: int = Field(description="Number of quantized layers")
+    num_parameters: int = Field(
+        description="Total number of all parameters (excluding biases)"
+    )
+    num_sparse_parameters: int = Field(
+        description="Total number of all sparsified parameters (excluding biases)"
+    )
+    num_four_blocks: int = Field(
+        description="Total number of all four blocks (excluding biases)"
+    )
+    num_sparse_four_blocks: int = Field(
+        description="Total number of all sparsified four blocks (excluding biases)"
+    )
+
+    nodes: List[NodeAnalysis] = Field(description="List of node analyses", default=[])
+
+    @classmethod
+    def from_onnx_model(cls, onnx_file_path: str):
+        """
+        Class constructor for Model Analysis
+
+        :param cls: class being constructed
+        :param onnx_file_path: path to onnx file being analyzed
+        :return: instance of ModelAnalysis class
+        """
+        model_onnx = onnx.load(onnx_file_path)
+
+        param_prunable_nodes = list(
+            filter(
+                partial(is_parameterized_prunable_layer, model_onnx),
+                model_onnx.graph.node,
+            )
+        )
+        sparsity_sizes = np.array(
+            [
+                get_node_num_zeros_and_size(model_onnx, node)
+                for node in param_prunable_nodes
+            ]
+        )
+        four_block_sparsity_sizes = np.array(
+            [
+                get_node_num_four_block_zeros_and_size(model_onnx, node)
+                for node in param_prunable_nodes
+            ]
+        )
+
+        num_sparse_parameters, num_parameters = np.sum(sparsity_sizes, axis=0)
+        num_sparse_four_blocks, num_four_blocks = np.sum(
+            four_block_sparsity_sizes, axis=0
+        )
+
+        layer_counts, op_counts = get_layer_and_op_counts(model_onnx)
+        num_nodes = len(model_onnx.graph.node)
+        num_layers = sum(layer_counts.values())
+        num_operations = sum(op_counts.values())
+        num_sparse_layers = map_and_accumulate(
+            model_onnx.graph.node, partial(is_sparse_layer, model_onnx), sum
+        )
+        num_quantized_layers = map_and_accumulate(
+            model_onnx.graph.node, partial(is_quantized_layer, model_onnx), sum
+        )
```

Comment:
similar to above, calling sum directly and using a comprehension will be a bit cleaner and readable.  I do like the helper function, just not a big value add here